src/hpr_routing.erl

```diff
@@ -52,8 +52,8 @@ handle_packet(Packet) ->
                     hpr_metrics:observe_packet_up(PacketType, ok, 0, Start),
                     ok;
                 {ok, Routes} ->
-                    Routed = maybe_deliver_packet(Packet, Routes, 0),
-                    ok = maybe_report_packet(Routes, Routed, Packet),
+                    {Routed, IsFree} = maybe_deliver_packet(Packet, Routes, 0, false),
```

Comment:
I think it's unlikely, but worth mentioning. 
There's a chance `IsFree` changes through the course of delivering the packet, and we'll only take the last value.

Reply:
yeah I know I think it's fine